src/include/migraphx/op/split.hpp

```diff
@@ -0,0 +1,213 @@
+                int axis_id   = 0;
+                int slice_dim = slice_dims[first];
+                for(auto&& len : input_shape.lens())
+                {
+                    if(axis_id == axis)
+                        out_dims.push_back(slice_dim);
+                    else
+                        out_dims.push_back(len);
+                    axis_id++;
+                }
+            }
+            else
+            {
+                std::vector<unsigned> slice_elements = compute_slice_elements(input_shape);
+                int total_elements                   = 0;
+                if(second >= slice_dims.size())
+                    MIGRAPHX_THROW("SPLIT:illegal split selector");
+
+                for(int i = first; i <= second; i++)
+                    total_elements += slice_elements[i];
+                if(total_elements <= 0)
+                    MIGRAPHX_THROW("SPLIT:invalid number of elements");
+
+                out_dims.push_back(total_elements);
+            }
+        }
+        else
+        {
+            out_dims.push_back(input_shape.elements());
+        }
+        return {input_shape.type(), out_dims};
+    }
+
+    std::vector<std::size_t> compute_index_map(const shape& s) const
+    {
+        std::vector<std::size_t> index_map;
+        std::size_t unit_slice = 1;
+        int axis_id            = 0;
+        if((axis < 0) || (axis >= s.lens().size()))
+            MIGRAPHX_THROW("SPLIT:invalid split axis");
+
+        for(auto&& len : s.lens())
+        {
+            if(axis_id++ > axis)
+                unit_slice *= len;
+        }
+        int total_slice_dim = 0;
+        for(auto&& dim : slice_dims)
+        {
+            if(dim == 0)
+                MIGRAPHX_THROW("SPLIT:invalid split dimension");
+            total_slice_dim += dim;
+        }
+        if(total_slice_dim != s.lens()[axis])
+            MIGRAPHX_THROW("SPLIT:invalid split dimension");
+
+        std::size_t stride    = unit_slice * total_slice_dim;
+        std::size_t nelements = s.elements();
+        std::vector<std::size_t> segment_size;
+        std::vector<std::size_t> begin_index;
+        std::size_t num_of_segments = nelements / stride;
+        std::size_t index           = 0;
+
+        // For each slice, compute segment size and begin index in the output.
+        for(auto&& dim : slice_dims)
+        {
+            std::size_t size = unit_slice * dim;
+            segment_size.push_back(size);
+            begin_index.push_back(index);
+            index += (num_of_segments * size);
+        }
+
+        // Map input index to output index.
+        index_map.resize(nelements);
+        for(std::size_t i = 0; i < nelements; i++)
+        {
+            std::size_t t_id          = i % stride;
+            std::size_t slice_id      = 0;
+            std::size_t element_index = 0;
+            std::size_t segment_id    = i / stride;
+            auto a_segment_size       = 0;
+            std::size_t id            = 0;
```

Comment:
Can you use the shape class to compute the indices?

Reply:
No.   This is the algorithm that I designed from scratch, nothing related to shape class.

Reply:
But the shape class already has methods to compute the index, why cant you re-use those methods?

Reply:
We are probably talking about two different things.  I am computing output index.   For each input index, the computation returns index in the big memory chunk, where spliteed results are concatted together in that memory.   I don't see how shape class supports this.

Reply:
Yes, you can create a shape for each slice you used to compute the index. Then the computation is just `s.index(i)+offset`.

Reply:
I don;t know what you are talking about.  Why don't you write the code?

Reply:
The "slice" here is different from the "slice" in your mind.  It is an algorithm I designed from scratch